exonum/src/runtime/mod.rs

```diff
@@ -303,15 +345,12 @@ pub trait Runtime: Send + Debug + 'static {
     /// Invoking of this callback is guaranteed to be the last operation for the runtime.
     /// Since this method is a part of shutdown process, runtimes can perform blocking and
     /// heavy operations here if needed.
-    fn shutdown(&self) {}
+    fn shutdown(&mut self) {}
```

Comment:
Maybe `on_shutdown` is more good name in according of `on_resume`?

Reply:
IMO, the naming is different on purpose, although I'm not 100% sure. When we run `Runtime::shutdown`, we *command* the runtime to shut down; there is no implied time dependencies between `shutdown` and other commands (other than that it runs after all other commands). When `Runtime::on_resume` is called, this has first and foremost *signalling* purpose; `Dispatcher` signals to the runtime that it has done restoring the state. It is not implied (at least currently) that the runtime should resume its state within the call.